detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt

```diff
@@ -287,17 +286,20 @@ class TorrentManager private constructor (
         strategies.seedingStrategy = strategyId
         strategies.storageLimit = storageLimit
 
-        val seedingTorrentsSorted = strategies.applyStrategy(strategyId, torrentFiles, profile.profiles)
+        val seedingTorrentsSorted = strategies.applyStrategy(
+            strategyId,
+            torrentFiles,
+            profile.profiles
+        ).distinctBy { it.handle }
         var storage: Long = 0
 
         val jobs = mutableListOf<Job>()
 
         val toStopSeeding = seedingTorrents.toMutableList()
         seedingTorrents.clear()
 
-
-        for (i in 0 until seedingTorrentsSorted.size) {
-            val size = seedingTorrentsSorted[i].handle.status().total() / 1000000 //TODO: store as byte to avoid all conversions
+        for (i in seedingTorrentsSorted.indices) {
+            val size = seedingTorrentsSorted[i].handle.status().total() / 1000000
```

Comment:
Not sure if you want to use a constant here and why the TODO got removed, I think it is also fine to leave as is though

Reply:
The constant is the conversion from bytes to mb, and I removed the todo because I doubt users will want to assign on bytes specific, in which case storing some extra zeros to store it in bytes doesn't make a lot of sense so figured it was better this way.